google/cloud/sql/connector/connector.py

```diff
@@ -321,6 +323,14 @@ async def connect_async(
         # attempt to make connection to Cloud SQL instance
         try:
             conn_info = await cache.connect_info()
+            # validate driver matches intended database engine
+            mapping = DriverMapping[driver.upper()]
+            if not conn_info.database_version.startswith(mapping.value):
+                raise IncompatibleDriverError(
+                    f"Database driver '{driver}' is incompatible with database "
```

Comment:
This is a fatal configuration error. I think the message could be more clear. Something like:

"Fatal Error: the Cloud SQL Connector is configured with an incompatible database driver. Configured driver '{driver}' cannot connect to database instance '{instance_connection_string}' running version {conn_info.database_version}."

Reply:
@hessjcg  I don't normally see Python errors get prefixed with "Fatal Error:", its more the assumption that if an error is raised from the application it must be fatal.

I like the error message as is personally, concise and actionable. I don't have access to `instance_connection_string` from where the error is raised after the encapsulation refactor...